src/function.cpp

```diff
@@ -51,19 +61,27 @@ DeclaredFunction DeclaredFunction::declare(std::vector<mlir::Type> &&domain,
   transform(domain.cbegin(), domain.cend(), back_inserter(smtDomain),
             getScalarSort);
 
-  if (range.isa<mlir::TensorType>()) {
-    throw UnsupportedException(
-        "Function that returns tensor is not supported yet");
-  }
-  if (range.isa<mlir::MemRefType>()) {
+  if (range.isIntOrIndexOrFloat()) {
+    const auto smtRange = getScalarSort(range);
+    FnDecl decl(smtDomain, smtRange, string(name) + "_tvfn");
+    return DeclaredFunction(move(domain), move(range), move(decl));
+  } else if (range.isa<mlir::TensorType>()) {
+    const auto tensorRange = range.dyn_cast<mlir::TensorType>();
```

Comment:
A ranked tensor might be dynamic: `tensor<?x4xi64>`.
Let's raise UnsupportedException in this case.

Reply:
Well, the code update below is checking whether the tensor is unranked...
There are three cases:
1. unranked tensor
2. ranked, dynamically sized tensor
3. ranked, statically sized tensor <-- the only case that must be allowed in this PR
I will update this.